features/ui/ui-kit-button/button.tsx

```diff
@@ -0,0 +1,90 @@
+import classNames from "classnames";
+import styles from "./button.module.scss";
+
+export type ButtonProps = {
+  children?: React.ReactNode;
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  as: "button" | "a";
+  href?: "string";
+  disabled?: boolean;
+  size: "small" | "medium" | "large" | "xlarge";
+  color:
+    | "primary"
+    | "secondary"
+    | "gray"
+    | "empty"
+    | "empty-gray"
+    | "error"
+    | "empty-error";
```

Comment:
Nit: In my implementation I split this up into `color` and `variant` as `empty` isn't really a color.